### DISCUSSION TITLE
Chaos Testing

### DISCUSSION THREAD

**Discussion by joshwoods** [MobileNativeFoundation/discussions#55] [General]
We recently discussed how our platform/backend teams handle Chaos Testing but I was really wondering what, if anything, other teams out there do to tackle this type of testing across mobile clients!?

**Comment by foobra** [comment]
consider `adb shell monkey` ?

**Comment by dlew** [comment]
On Android, we have a chaos monkey built into our OkHttp layer as an Interceptor. It adds extra time to network calls randomly, and it will occasionally fail them (sometimes before the call is made, sometimes afterwards to simulate us not receiving the response). It's configurable via a debug menu.

It is very useful for us, especially since we support offline mode; it's good to know that things retry successfully even when things fail.
